src/Microsoft.TestPlatform.Common/Utilities/FakesUtilities.cs

```diff
@@ -50,17 +52,44 @@ public static string GenerateFakesSettingsForRunConfiguration(string[] sources,
                 doc.Load(xmlReader);
             }
 
-            return !TryAddFakesDataCollectorSettings(doc, sources, GetFramework(runSettingsXml)) 
-                ? runSettingsXml 
-                : doc.OuterXml;
+            var frameworkVersion = GetFramework(runSettingsXml);
+            if (frameworkVersion == null)
+            {
+                return runSettingsXml;
```

Comment:
it would be nice to add logs in this case. I dont think the EqtTraceLogs go anywhere useful today. We'd probably need a logger interface passed through. To reduce code churn in a release branch, we could do that in the master branch as a follow up change.